src/Transactions.sol

```diff
@@ -22,7 +22,22 @@ library Transactions {
         bytes v;
     }
 
-    function encodeRLP(Legacy memory txStruct) internal pure returns (bytes memory) {
+    struct EIP1559 {
+        address to;
+        uint64 gas;
+        uint64 maxFeePerGas;
+        uint64 maxPriorityFeePerGas;
+        uint64 value;
+        uint64 nonce;
+        bytes data;
+        uint64 chainId;
+        bytes[] accessList;
+        bytes r;
+        bytes s;
+        bytes v;
+    }
+
+    function encodeLegacyRLP(Legacy memory txStruct) internal pure returns (bytes memory) {
```

Comment:
Can you keep the same name here? I imagine having the same `encodeRLP` name overloaded for both transactions.